src/rules/exports-last.js

```diff
@@ -0,0 +1,35 @@
+function isExportStatement({ type }) {
+  // ES Module export statements
+  if (type === 'ExportDefaultDeclaration' || type === 'ExportNamedDeclaration') {
+    return true
+  }
+
+  return false
+}
+
+const rule = {
+  create(context) {
+    return {
+      Program({ body }) {
+        const lastNonExportStatement = body.reduce((acc, node, index) => {
+          if (isExportStatement(node)) {
+            return acc
+          }
+          return index
+        }, 0)
+
+        body.forEach((node, index) => {
+          if (isExportStatement(node) && index < lastNonExportStatement) {
```

Comment:
It might be a bit faster to do the `index < lastNonExportStatement` before the `isExportStatement`.

Or even better (?): only run the `forEach` on the statements after `lastNonExportStatement`.

```js
// maybe add +/-1 to lastNonExportStatement to make it work right
body.slice(lastNonExportStatement).forEach(node => {
  if (isExportStatement(node)) {
    // report
  }
})
```